src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -296,8 +296,13 @@ public AzureAppConfigurationOptions ConfigureKeyVault(Action<AzureAppConfigurati
             var keyVaultOptions = new AzureAppConfigurationKeyVaultOptions();
             configure?.Invoke(keyVaultOptions);
 
+            if (keyVaultOptions.Credential != null && keyVaultOptions.SecretResolver != null)
+            {
+                throw new InvalidOperationException($"Cannot configure both default credentials and secret resolver for Key Vault references. Please call either {nameof(keyVaultOptions.SetCredential)} or {nameof(keyVaultOptions.SetSecretResolver)} method, not both.");
```

Comment:
Why are we throwing this here instead of in the options?

Reply:
Oh apparently I suggested this in a comment. Sorry, I should have suggested that it be in `AzureAppConfigurationKeyVaultOptions`

Reply:
Now that I look at this again, having it here is even better than what I was originally thinking. Resolving this comment.

Reply:
In `AzureAppConfigurationKeyVaultOptions`, we'd need to add this check in 2 places - `SetCredential` and `SetSecretResolver` (as opposed to just one check in `ConfigureKeyVault`).

Reply:
That's true, but that's not the reasoning that sold me on this. I like how with the way you have it now that `AzureAppConfigurationKeyVaultOptions` can be a POCO and `ConfigureKeyVault` can be smart and contain the business logic. Having simple objects is something I find valuable.